### PR TITLE
Adds the handheld syndicate teleporter

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds the handheld syndicate teleporter, 8 tc, 4 charges, 10% chance to recharge 1 charge each cycle, teleports user 4-8 tiles forward, if chosen destination is a wall, attempt an emergency teleport 3 tiles parallel to destination, if successful, go there, if fail, gibs the user. Force teleports with no safety on EMP, or teleports out of existence if not in a persons inventory. If a BOH is on the teleporter, teleports randomly with no saftey. If you teleport onto a person (1/4 chance if lined up right) deals 20 brute to them, and stuns for 3. High risk, high reward.

## Why It's Good For The Game

An actual mobility item for traitors is good, to bypass walls / doors, in a semi controlled but still risky way. Not overly abusable, spamming will leave you out of charges at best, and a pile of gore at worst. Unlike viel shifter, you can not pull people with you, no teleporting on captain, stunning, teleporting out with them. If you use the item right, there is almost no risk of you messing up, but quick uses may punish you. Has counters / prevention like most teleporting equipment, with EMP automatically triggering it, a good way to deal with a user of it, and preventing emp bombing the AI easily, and like a normal teleporter, not working well with a bag of holding.

What if nukies use it?
Can be removed from nukies if it proves to be problematic, and even if they teleport into a barricaded bridge, they need to spend time stripping captain, and should they use an EMP at a bad time, or use it in the wrong spot... splat.

##TODO

- [x] Make the description for the item better

- [x] Make it not work in mech / locker / sleeper / ect

## Images of changes
Normal action. 
https://imgur.com/a/AhdURck

EMP_act
https://imgur.com/a/MwJuDf3

Gibbing.
https://imgur.com/a/NsiIMkk

## Changelog
:cl:
add: Adds the syndicate teleporter, an experimental teleporter syndicate agents can buy.
/:cl:

### CODE REVIEW

**Review by Kyep — APPROVED** [review]

**Comment by Denthamos** [comment]
Oh god, the baldie agents using this gibbing is going to be hilarious.

**Comment by Qwertytoforty** [comment]
aaaaand it auto requested review from code owners

**Comment by Qwertytoforty** [comment]
I'll just figure this out when I am not busy, please don't shoot me, shouldnt be 520 files changed now

File: README.md [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
-`Paradise/icons/goonstation`, or
-`Paradise/sound/goonstation`
-directories, or any subdirectories of mentioned directories are licensed under the
-Creative Commons 3.0 BY-NC-SA license
-(https://creativecommons.org/licenses/by-nc-sa/3.0)
-
-All other assets including icons and sound files are licensed under the
-Creative Commons 3.0 BY-SA license (https://creativecommons.org/licenses/by-sa/3.0/),
-unless otherwise indicated.
+# Paradise
```
Comment (AffectedArc07) [inline README.md:1]: You goofed the line endings here

File: code/game/objects/items/devices/traitordevices.dm [hunk 2]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+/obj/item/teleporter/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>[src] has [charges] out of [max_charges] charges left.</span>"
+
+/obj/item/teleporter/attack_self(mob/user)
+	attempt_teleport(user, FALSE)
+
+/obj/item/teleporter/process()
+	if(prob(10) && charges != max_charges)
+		charges += 1
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:208]:
```suggestion
	if(prob(10) && charges < max_charges)
		charges++
```
Styling + This prevents it being infinitely overcharged if it somehow goes above

File: code/game/objects/items/devices/traitordevices.dm [hunk 3]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+
+/obj/item/teleporter/emp_act(severity)
+	if(prob(50 / severity))
+		var/mob/living/carbon/human/user = src.loc
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:212]:
```suggestion
		var/mob/living/carbon/human/user = loc
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 4]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+		if((user && ishuman(user)))
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:213]:
```suggestion
		if(user && ishuman(user))
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 5]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+			to_chat(user, "<span class='danger'>The [src] buzzes and activates!</span>")
+			attempt_teleport(user, TRUE)
+		else
+			src.visible_message("<span class='warning'> The [src] activates and blinks out of existance!</span>")
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:217]:
```suggestion
			visible_message("<span class='warning'> The [src] activates and blinks out of existance!</span>")
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 6]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+
+	var/mob/living/carbon/C = user
+	var/turf/mobloc = get_turf(C)
+	var/list/turfs = new/list()
+	var/found_turf = FALSE
+	var/list/bagholding = user.search_contents_for(/obj/item/storage/backpack/holding)
+	for(var/turf/T in range(user, tp_range))
+		if(!is_teleport_allowed(T.z))
+			break
+		if(!(bagholding.len && !flawless)) //Chaos if you have a bag of holding
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:234]:
```suggestion
		if(!(length(bagholding) && !flawless)) //Chaos if you have a bag of holding
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 7]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+
+		turfs += T
+		found_turf = TRUE
+
+	if(found_turf)
+		if(user.loc != mobloc) // No locker / mech / sleeper teleporting, that breaks stuff
+			to_chat(C, "<span class='danger'>The [src] will not work here!</span>")
+		charges--
+		var/turf/destination = pick(turfs)
+		var/turf/TC = destination
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:252]: Why make this another variable?

File: code/game/objects/items/devices/traitordevices.dm [hunk 8]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+			get_fragged(user, destination)
+	else
+		to_chat(C, "<span class='danger'>The [src] will not work here!</span>")
+
+/obj/item/teleporter/proc/tile_check(turf/TC)
+	if(istype(TC, /turf/simulated/floor) || istype(TC, /turf/space) || istype(TC, /turf/simulated/shuttle/floor) || istype(TC, /turf/simulated/shuttle/floor4) || istype(TC, /turf/simulated/shuttle/plating))
+		return TRUE
+
+/obj/item/teleporter/proc/panic_teleport(mob/user, turf/destination, direction = NORTH)
+	var/saving_throw = 0
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:274]:
```suggestion
	var/saving_throw
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 9]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+			if(prob(50))
+				saving_throw = NORTH
+			else
+				saving_throw = SOUTH
+		else
+			saving_throw = NORTH // just in case
+
+	var/mob/living/carbon/C = user
+	var/turf/mobloc = get_turf(C)
+	var/list/turfs = new/list()
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:291]:
```suggestion
	var/list/turfs = list()
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 10]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+		found_turf = TRUE
+
+	if(found_turf)
+		var/turf/new_destination = pick(turfs)
+		var/turf/fragging_location = new_destination
+		telefrag(fragging_location, user)
+		C.forceMove(new_destination)
+		playsound(mobloc, "sparks", 50, TRUE)
+		new/obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+		new/obj/effect/temp_visual/teleport_abductor/syndi_teleporter(new_destination)
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:313]:
```suggestion
		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(new_destination)
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 11]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+	else //We tried to save. We failed. Death time.
+		get_fragged(user, destination)
+
+
+/obj/item/teleporter/proc/get_fragged(mob/user, turf/destination)
+	var/turf/mobloc = get_turf(user)
+	user.forceMove(destination)
+	playsound(mobloc, "sparks", 50, TRUE)
+	new/obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+	new/obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:324]:
```suggestion
	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 12]
```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
...
+"}
+/obj/item/storage/box/syndie_kit/teleporter
+	name = "syndicate teleporter kit"
+
+/obj/item/storage/box/syndie_kit/teleporter/New()
+	..()
+	new /obj/item/teleporter(src)
+	new /obj/item/paper/teleporter(src)
+	return
+/obj/effect/temp_visual/teleport_abductor/syndi_teleporter
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:364]:
```suggestion
	return

/obj/effect/temp_visual/teleport_abductor/syndi_teleporter
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Qwertytoforty) [reply README.md]: fixed

File: code/game/objects/items/devices/traitordevices.dm (hunk 7 above)
Comment (Qwertytoforty) [reply code/game/objects/items/devices/traitordevices.dm]: Removed

File: code/game/objects/items/devices/traitordevices.dm [hunk 13]
```diff
@@ -266,8 +265,8 @@ effective or pretty fucking useless.
 	else
 		to_chat(C, "<span class='danger'>The [src] will not work here!</span>")
 
-/obj/item/teleporter/proc/tile_check(turf/TC)
-	if(istype(TC, /turf/simulated/floor) || istype(TC, /turf/space) || istype(TC, /turf/simulated/shuttle/floor) || istype(TC, /turf/simulated/shuttle/floor4) || istype(TC, /turf/simulated/shuttle/plating))
+/obj/item/teleporter/proc/tile_check(turf)
+	if(istype(turf, /turf/simulated/floor) || istype(turf, /turf/space) || istype(turf, /turf/simulated/shuttle/floor) || istype(turf, /turf/simulated/shuttle/floor4) || istype(turf, /turf/simulated/shuttle/plating))
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:269]:
```suggestion
/obj/item/teleporter/proc/tile_check(turf/T)
	if(istype(T, /turf/simulated/floor) || istype(T, /turf/space) || istype(T, /turf/simulated/shuttle/floor) || istype(T, /turf/simulated/shuttle/floor4) || istype(T, /turf/simulated/shuttle/plating))
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/traitordevices.dm [hunk 14]
```diff
@@ -168,3 +168,203 @@ effective or pretty fucking useless.
...
+/obj/item/teleporter/attack_self(mob/user)
+	attempt_teleport(user, FALSE)
+
+/obj/item/teleporter/process()
+	if(prob(10) && charges < max_charges)
+		charges++
+
+/obj/item/teleporter/emp_act(severity)
+	if(prob(50 / severity))
+		var/mob/living/carbon/human/user = loc
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:212]: How will this handle if the tele is on the floor and its EMP'd? Or if its inside a closet. Its not always going to be help, so I would move this check elsehwere.

File: code/game/objects/items/devices/traitordevices.dm [hunk 15]
```diff
@@ -168,3 +168,203 @@ effective or pretty fucking useless.
...
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+	playsound(destination, "sparks", 50, TRUE)
+	playsound(destination, "sound/magic/disintegrate.ogg", 50, TRUE)
+	destination.ex_act(rand(1,2))
+	for(var/obj/item/W in user)
+		if(istype(W, /obj/item/organ)|| istype(W, /obj/item/implant))
+			continue
+		if(!user.unEquip(W))
+			qdel(W)
+	to_chat(user, "<span class='userdanger'><font size=3>You teleport into the wall, the teleporter tries to save you, but--</font></span>")
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:331]: `biggerdanger` may be a better class here, then you dont have to stack tags.

File: code/game/objects/items/devices/traitordevices.dm [hunk 16]
```diff
@@ -168,3 +168,203 @@ effective or pretty fucking useless.
...
+	..()
+	new /obj/item/teleporter(src)
+	new /obj/item/paper/teleporter(src)
+	return
+
+/obj/effect/temp_visual/teleport_abductor/syndi_teleporter
+	duration = 5
+
+/obj/item/teleporter/admin
+	desc = "A strange syndicate version of a cult veil shifter. \n\ This one seems EMP proof, and with much better saftey protocols."
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/traitordevices.dm:367]: `\n\`

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/traitordevices.dm (hunk 15 above)
Comment (Qwertytoforty) [reply code/game/objects/items/devices/traitordevices.dm:331]: Done

File: code/game/objects/items/devices/traitordevices.dm (hunk 16 above)
Comment (Qwertytoforty) [reply code/game/objects/items/devices/traitordevices.dm]: Done

File: code/game/objects/items/devices/traitordevices.dm (hunk 14 above)
Comment (Qwertytoforty) [reply code/game/objects/items/devices/traitordevices.dm]: moved!

**Pull request merged**